mam/v2/mss.c

```diff
@@ -276,8 +273,8 @@ void mss_init(mss_t *const mss, prng_t *const prng, sponge_t *const sponge,
   mss->prng = prng;
   mss->sponge = sponge;
   mss->wots = wots;
-  mss->nonce1 = nonce1;
-  mss->nonce2 = nonce2;
+  trit_array_set_trits(&mss->nonce1, nonce1->trits, nonce1->num_trits);
```

Comment:
This relies on the value of `mss->nonce1.dynamic` to do a free or not.
Since it hasn't been initialised yet, the value of `dynamic` is random and it will most probably attempt a `free` on an invalid pointer.